site2/tools/apidoc/python/pydoctor-generator-helper.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+
+set -x -e
+
+if [ -z "$PULSAR_VERSION" ]; then
+    echo "PULSAR_VERSION must be set."
+    exit 1
+fi
+
+python -m pip install --upgrade pip
+
+pip install -U pydoctor
+git clone https://github.com/apache/pulsar-client-python --single-branch --branch v$PULSAR_VERSION
```

Comment:
cc @merlimat @BewareMyPower @RobertIndie this line depends on that for each Python client release, we create a `vX.Y.Z` tag. I notice that C++ client does it so make this assumption.

Reply:
Verified the overall script by hardcoding the tag as `main` locally.